### PR TITLE
STR-676: sets a CI autoupdate cron job

### PR DESCRIPTION
**Pull request by storopoli**
## Description

Spinoff of #537.

Adds a CI cron job to do an automated PR monthly.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature/Enhancement (non-breaking change which adds functionality or enhances an existing one)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactor
- [ ] New or updated tests
- [ ] Dependency Update
- [x] New CI action

## Notes to Reviewers

Depends on `https://github.com/organizations/alpenlabs/settings/actions` and check `Allow GitHub Actions to create and approve pull requests`.

If you look closely, you might find a Ramanujan number hidden in this PR.

## Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented my code where necessary.
- [x] I have updated the documentation if needed.
- [x] My changes do not introduce new warnings.
- [x] I have added tests that prove my changes are effective or that my feature works.
- [x] New and existing tests pass with my changes.

## Related Issues

STR-676
STR-630

### CODE REVIEW

**Review by delbonis — APPROVED** [review]

File: .github/workflows/cron-nightly-rust.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    name: Update nightly Rustc
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - uses: dtolnay/rust-toolchain@nightly
+      - name: Update rust-toolchain.toml to use latest nightly
+        run: |
+          set -x
```
Comment (Rajil1213) [inline .github/workflows/cron-nightly-rust.yml:19]: Should fail on error as a best-practice.

```suggestion
          set -xe
```

File: .github/workflows/cron-nightly-rust.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
+name: Update nightly Rust
+on:
+  schedule:
+    - cron: "29 17 1 * *" # At 17:29 on day-of-month 1.
```
Comment (Rajil1213) [inline .github/workflows/cron-nightly-rust.yml:4]: Why this specific time?

**Review by Rajil1213 — CHANGES_REQUESTED** [review]
Left a few nits.

**Comment by delbonis** [comment]
Agree with ^comments.

File: .github/workflows/cron-nightly-rust.yml (hunk 2 above)
Comment (Rajil1213) [reply .github/workflows/cron-nightly-rust.yml:4]: Ohkayy. So that's the Ramanujan number from the PR description. I guess it's fine. 🤷

Comment (storopoli) [reply .github/workflows/cron-nightly-rust.yml:4]: Yes also imagine how many stuff around the world runs at 0h UTC.

**Review by Rajil1213 — APPROVED** [review]

**Pull request merged**